projects/plugins/boost/app/modules/optimizations/minify/Minify.php

```diff
@@ -24,49 +21,33 @@ public function setup() {
 			return;
 		}
 
-		add_action( Config::get_cron_cache_cleanup_hook(), 'jetpack_boost_page_optimize_cache_cleanup' );
-		register_deactivation_hook( JETPACK_BOOST_PATH, 'jetpack_boost_page_optimize_deactivate' );
-		register_uninstall_hook( JETPACK_BOOST_PATH, 'jetpack_boost_page_optimize_uninstall' );
+		if ( ! self::$called ) {
+			add_action( Config::get_cron_cache_cleanup_hook(), 'jetpack_boost_page_optimize_cache_cleanup' );
+			register_deactivation_hook( JETPACK_BOOST_PATH, 'jetpack_boost_page_optimize_deactivate' );
+			register_uninstall_hook( JETPACK_BOOST_PATH, 'jetpack_boost_page_optimize_uninstall' );
 
-		jetpack_boost_page_optimize_schedule_cache_cleanup();
+			jetpack_boost_page_optimize_schedule_cache_cleanup();
+		}
 
 		add_action( 'init', array( $this, 'init_concatenate' ) );
 
-		// Disable Jetpack photon-cdn for static JS/CSS
-		add_filter( 'jetpack_force_disable_site_accelerator', '__return_true' );
-	}
+		if ( ! self::$called ) {
+			// Disable Jetpack photon-cdn for static JS/CSS
+			add_filter( 'jetpack_force_disable_site_accelerator', '__return_true' );
+		}
 
-	public static function get_slug() {
-		return 'minify';
+		self::$called = true;
 	}
 
 	public static function is_available() {
-		return defined( 'JETPACK_BOOST_MINIFY' ) && true === JETPACK_BOOST_MINIFY;
+		return true;
 	}
 
 	public function setup_trigger() {
 		return 'init';
 	}
 
 	public function init_concatenate() {
-		if ( is_admin() ) {
-			return;
-		}
-
-		jetpack_boost_init_filesystem();
-
-		if ( jetpack_boost_page_optimize_should_concat_js() || jetpack_boost_page_optimize_load_mode_js() ) {
-			global $wp_scripts;
-
-			$wp_scripts                         = new Concatenate_JS( $wp_scripts );
-			$wp_scripts->allow_gzip_compression = true; // @todo - used constant ALLOW_GZIP_COMPRESSION = true if not defined.
-		}
-
-		if ( jetpack_boost_page_optimize_should_concat_css() ) {
-			global $wp_styles;
-
-			$wp_styles                         = new Concatenate_CSS( $wp_styles );
-			$wp_styles->allow_gzip_compression = true; // @todo - used constant ALLOW_GZIP_COMPRESSION = true if not defined.
-		}
+		// declared in child classes.
```

Comment:
If it's absolutely necessary to create extendable classes that require a method, this should be declared as an abstract class that forces extenders to declare `init_concatenate`